GoogleForCreators/web-stories-wp
# Media: Fix FFmpeg memory leaks

## Context



There is an issue with memory consumption when invoking ffmpeg.wasm, where the worker threads are not properly closed by ffmpeg.wasm itself and accumulating with each use.

## Summary



This PR fixes ffmpeg.wasm by completely exiting after file processing is done.

## Relevant Technical Choices



* Refactor `useFFmpeg` a little bit to better reuse FFmpeg config options
* Add `console.error()` statements so that errors using ffmpeg.wasm are not secretly discarded, making debugging easier.
* Moves the `exit()` call to the `finally` block so that it runs no matter what.
* Wraps the `exit()` call in a try catch to prevent the errors from bubbling up

## To-do



## User-facing changes



N/A

## Testing Instructions






This PR can be tested by following these steps:

1. Open Chrome's Task Manager and the "Sources" tab in the dev tools.
2. Upload a video to the canvas so it gets transcoded
3. See a bunch of worker threads being created and memory usage to increase significantly
4. Wait for upload to finish
5. See worker threads being removed and memory decreasing again to previous levels

## Reviews

### Does this PR have a security-related impact?



No

### Does this PR change what data or activity we track or use?



No

### Does this PR have a legal-related impact?



No

## Checklist




---



Fixes #8768
